### ISSUE TITLE
Add parsing

### ISSUE THREAD

**Issue by dodekeract** [dodekeract/spirit-body#1]
- [ ] JSON
- [ ] Form

**Comment by hnry** [comment]
Hey @dodekeract , if I can add some suggestions:
- Should check for size, if the body is huge, give up reading and continue the request.
Have it be programmatically checkable. Maybe `request.body` is a `Error` object ?

- I think it'll be better to use existing libraries to parse the body
https://www.npmjs.com/package/qs
or `querystring` from nodejs core are some suggestions

- I think it's better to write a generic module over targeting spirit directly?
For example, https://www.npmjs.com/package/raw-body is the module all the koa, express middlewares use. The middlewares itself is very light and simply just calls raw-body. I was just thinking of doing the same with spirit, but never got around to it.

**Comment by dodekeract** [comment]
@hnry I'm aware of these issues, originally this project was just meant to be a simple JSON parser middleware. 🙂

Then @fvj implemented URLEncoding.

I agree on using existing libraries, always planned to do that sooner or later. Especially multipart is quite hard to parse properly...

> I think it's better to write a generic module over targeting `spirit` directly. — @hnry

While I agree in most cases - there are already a lot of generic modules that do this. The goal of this project is (now) to provide a trivially easy-to-use middleware for spirit. Without using `spirit-express`, because I don't really like `spirit-express` TBH.

When I find some time I'll definitely look into using something like `qs` or `raw-body` in `spirit-body`.
